util.ts

```diff
@@ -0,0 +1,44 @@
+import zlib = require('zlib');
+import {IHttpClientResponse} from './interfaces'
+
+/**
+ * Decompress/Decode gzip encoded JSON
+ * Using Node.js built-in zlib module
+ *
+ * @param {Buffer} buffer
+ * @param {string} charset
+ * @return {Promise<string>}
+ */
+export async function decompressGzippedContent(buffer: Buffer, charset: string): Promise<string> {
+    return new Promise<string>(async (resolve, reject) => {
```

Comment:
nit: I don't think you need the `async` keyword if you're explicitly returning a promise and not using the `await` keyword within the function's body (not counting inline functions within that). I think same applies to use of `async` in the other file.

```suggestion
export function decompressGzippedContent(buffer: Buffer, charset: string): Promise<string> {
    return new Promise<string>((resolve, reject) => {
```